### DISCUSSION TITLE
dde-file-manager compiled from source fails at runtime with undefined symbol error

### DISCUSSION THREAD

**Discussion by DonaldKellett** [linuxdeepin/developer-center#3484] [Development Questions | 开发答疑] [answered]
Hi there, I recently forked dde-file-manager on GitHub and cloned the fork locally, then installed the build dependencies and compiled the project from source according to the instructions in the README. The compile step succeeded, but now, when I run the compiled binary `$PROJECT_ROOT/build/src/dde-file-manager/dde-file-manager`, I get the following error about an undefined symbol:

```
./src/dde-file-manager/dde-file-manager: symbol lookup error: ./src/dde-file-manager/dde-file-manager: undefined symbol: _ZN4RLog6commitERK7QStringRK4QMapIS0_8QVariantE
```

A quick search suggests that my DTK version might not be new enough and needs updating. Is that indeed the cause of the problem, and if so, how do I update my DTK? Thanks!

OS: deepin 20.7.1 Community, deepin 23 Preview (exact same error happens on both, down to the name of the undefined symbol)

Related (Discord): https://discord.com/channels/851251275151704124/856337463999070229/1038085497303601232

**Comment by BLumia** [comment, marked as answer]
Hi,

Hacking on DDE apps may need to use the newest version of DTK. If you are trying to fix an existing bug, it's recommended to switch to the tag version that your distro is currently using. If the tag version still has similar issues, please feel free to let us know.

If you intended to build the git HEAD version, then you probably want to build and install the git HEAD version of DTK first. V23 is still in a preview stage so a developer repo is still not available for now, we might provide a develop repo in later updates :)

btw, you can join https://t.me/deepin_community (currently most members are Chinese) or https://t.me/deepin for developer related questions. We also have a [mailing list](https://www.freelists.org/archive/deepin-devel/) for developer-related topic in case you are interested :)

**Reply by DonaldKellett** [reply]
Thanks for the response, I do intend to build and hack on the latest development snapshot of dde-file-manager (and possibly other DDE system apps down the line). By the git HEAD version of DTK, do you mean [DtkCore](https://github.com/linuxdeepin/dtkcore), or is there more to it than just that?

Also, thanks for the invite! I don't currently have a Telegram account but I'll consider creating one. The mailing list also seems interesting.

**Reply by BLumia** [reply]
You might need both `dtkcommon`, `dtkcore`, `dtkgui`, and `dtkwidget`. `dtkcommon` should be the first one to build & install and `dtkwidget` should be the last one. I can't remember if you need to build & install `dtkgui` first or build & install `dtkcore` first but I think you might figure it out easily while building.

> I don't currently have a Telegram account but I'll consider creating one.

These Telegram groups have Matrix bridge enabled so you can join the matrix side of the chat ([#deepin-community:matrix.org](https://matrix.to/#/#deepin-community:matrix.org) and [#deepin:matrix.org](https://matrix.to/#/#deepin:matrix.org)) if you have a Matrix account and don't want to register Telegram :)

**Reply by DonaldKellett** [reply]
Thanks for the pointers. Unfortunately, even after building + installing the packages for the latest development snapshots of `dtkcommon`, `dtkcore`, `dtkgui` and `dtkwidget` (in that order), dde-file-manager still complains about the exact same undefined symbol after successful compilation of the project using `make` :cry:

This was on deepin 20.7.1 Community, but I suspect I'll get the same result if I try it on deepin 23 Preview as well. I've joined the Telegram group last night and will probably ask them for any last ideas to rectify the issue, so thanks for the invitation!

**Reply by BLumia** [reply]
After looking at `c++filt _ZN4RLog6commitERK7QStringRK4QMapIS0_8QVariantE`, the symbol is provided by `dde-file-manager-lib`, so your issue is caused by not using the `libdde-file-manager.so` that you built. It can be resolved by setting `LD_LIBRARY_PATH` to correct places.

For convenience, I recommended using Qt Creator to open filemanager.pro project file, then the Environment Variables will be taken care of by the IDE so you can start hacking it without issue.

**Reply by DonaldKellett** [reply]
Thanks, the issue was indeed resolved after I installed Qt Creator and used it to build the project :heart:
